csrc/runtime/fusion_kernel_runtime.cpp

```diff
@@ -597,14 +597,41 @@ std::optional<std::unique_ptr<HeuristicParamsList>> FusionKernelRuntime::
       heuristics->at(group_to_run->groupId()) = std::move(heuristic_params);
     }
 
-    // Generate metadata for the fusion's outputs
-    auto group_runtime_outputs = inferOutputSizes(
-        fusion_to_run,
-        group_runtime_inputs,
-        evaluator_precomputed_values.get());
+    // Generate metadata for the fusion's outputs. If these outputs are
+    // generated by NVFuser fusions, we assume the outputs are contiguous, and
+    // compute the sizes and strides based on this assumption. If these outputs
+    // are generated by ExpressionEvaluator, we use tensors on meta device and
+    // actually invoking ExpressionEvaluator to compute the sizes and strides.
+    KernelArgumentHolder group_runtime_outputs;
+    const auto& heuristic_params = heuristics->at(group_to_run->groupId());
+    const bool is_expr_eval =
+        heuristic_params->scheduler_type == SchedulerType::ExprEval;
+    if (is_expr_eval) {
+      // For expr evaluated fusion, the striding rules follow that of ATen.
```

Comment:
This is great! Essentially, the code evaluates **meta** tensors to get the correct strides. 

Can we somehow move this into inferShapeOfOutput? For example, inferShapeOfOutput is used in https://github.com/NVIDIA/Fuser/blob/524dcf12dba1a4dc1318a1bcc60b6e45ff4c8d3e/csrc/ir/nodes.cpp#L4762C34-L4762C52. I believe right now it always returns contiguous strides due to https://github.com/NVIDIA/Fuser/blob/524dcf12dba1a4dc1318a1bcc60b6e45ff4c8d3e/csrc/runtime/allocations.cpp#L193, which is a bug.

Reply:
Is `inferShapeOfOutput` the unified function that all output shape/stride inference goes here. It would be nice to infer the correct value in the first place, instead of inferring a wrong one and patching it with `resetAllocationDomainAndContiguity`